cmd/torpedo/torpedo.go

```diff
@@ -592,39 +593,74 @@ func (t *torpedo) run(testName string) error {
 }
 
 func main() {
-	// TODO: switch to a proper argument parser
-	if len(os.Args) < 3 {
-		logrus.Infof("Usage: %v <scheduler> <volume driver> <node driver> [testName]", os.Args[0])
-		os.Exit(-1)
-	}
 
-	testName := "" // TODO csv or yaml based args
-	if len(os.Args) > 4 {
-		testName = os.Args[4]
+	app := cli.NewApp()
+	app.Name = "torpedo"
+	app.Usage = "Run the torpedo scheduler test suite."
+
+	app.Commands = []cli.Command{
+		{
+			Name:    "fire",
+			Aliases: []string{"f"},
+			Usage:   "Starts the basic test suite",
+			Action:  fireTorpedo,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "scheduler,s",
+					Usage: "Name of the scheduler to use",
+					Value: "k8s",
+				},
+				cli.StringFlag{
+					Name:  "storage,v",
+					Usage: "Name of the storage driver to use",
+					Value: "pxd",
+				},
+				cli.StringFlag{
+					Name:  "node-driver,n",
+					Usage: "Name of the node driver to use",
+					Value: "ssh",
+				},
+				cli.StringFlag{
+					Name:  "test,t",
+					Usage: "Name of the test to run",
+					Value: "all",
+				},
+			},
+		},
 	}
+	app.Run(os.Args)
+
+}
+
+func fireTorpedo(c *cli.Context) {
+	s := c.String("scheduler")
```

Comment:
Better if the cli flags like "scheduler" are constants that are also referenced while creating the cli flags above. So if someone changes one, other changes automatically.

Reply:
addressed in subsequent commit.